bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BookieRequestProcessor.java

```diff
@@ -342,7 +361,15 @@ private void processReadRequestV3(final BookkeeperProtocol.Request r, final Chan
             }
         } else {
             read = new ReadEntryProcessorV3(r, c, this, fenceThreadPool);
-            if (null == readThreadPool) {
+
+            // If it's a recovery read, we want to make sure it gets submitted right away,
+            // bypassing the eventual executor queue,
+            // to reduce the chances of ledger recovery exceptions.
+            boolean isRecoveryRead = r.getReadRequest().hasFlag()
```

Comment:
Can we make this into some until function and have a test case cover it?

Reply:
Will do.